codegen/src/main/java/org/nd4j/codegen/cli/CLI.java

```diff
@@ -19,13 +21,76 @@
 public class CLI {
     private static final String relativePath = "nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/";
 
-
     @Parameter(names = "-dir", description = "Root directory of deeplearning4j mono repo", required = true)
     private String repoRootDir;
 
     @Parameter(names = "-namespaces", description = "List of namespaces to generate, or 'ALL' to generate all namespaces", required = true)
     private List<String> namespaces;
 
+    @Parameter(names = "-projects", description = "List of sub-projects - ND4J, SameDiff or both", required = false)
+    private List<String> projects;
```

Comment:
FYI I tested this on the command line - looks like we don't need to update generate.sh if we do this:
`./generate.sh image -projects sd`
We'll need to update the readme.md file eventually too.

Reply:
Readme updated. Extra argument is passed through correctly from last position.